### ISSUE TITLE
Bundle size

### ISSUE THREAD

**Issue by wwwater** [onfido/onfido-sdk-ui#2143]
Hi,

we've noticed that the bundle size has increased significantly over the years. A few years back it was 2 MB, today it's 10 MB for us. Do you have plans in reducing it? Is there a way for us to reduce it (we use webpack and include it in the code via `import`)?

**Comment by warrenseine** [comment]
The bundle itself is ~2.5 MB, or less than 500 kB compressed. It can fetch additional resources dynamically, but that shouldn't impact your application bundle. How did you estimate these 10 MB?

**Comment by wwwater** [comment]
@warrenseine That's the size bundle produced by webpack.

A few releases back we had a problem that it wasn't building because of missing nodejs polyfills, so we asked (https://github.com/onfido/onfido-sdk-web-sample-app/issues/129) and got a reply that some of the dependencies aren't actually needed. Thus I was wondering whether we could reduce the fiinal size more, I'm not an expert on JS packaging.

**Comment by warrenseine** [comment]
There was a bug introduced a specific release that made it look like it needed the `fs` package and we fixed it in a patch version, but we didn't really remove anything. There's no package we could remove for now. I think the small increase we're seeing is mostly organic due to new features.

We do have a long-term plan to make the SDK more modular, which could help with the bundle size, but nothing that can help right now.

I'm still curious about the bundle size you're reporting. If Webpack is producing 10 MB bundle, then you're adding something. Could it be other packages? Or source maps?

**Comment by giomogna** [comment]
I can confirm that the bundle size is ~10MB for us too (also on [bundlephobia](https://bundlephobia.com/package/onfido-sdk-ui@10.4.0)). 

<img width="1731" alt="Screenshot 2023-02-08 at 15 01 16" src="https://user-images.githubusercontent.com/71967930/217551439-b17b13bf-4fba-42c3-a116-bb59cd995b11.png">

That's the total size, dependencies (like `@onfido/active-video-capture`) included.
The size of `onfido-sdk-ui` alone (as shown on bundlephobia) is ~4.13MB, which is roughly the size of `/lib/index.js`.

2.5MB seems to be the size of `/dist/onfido.min.js`, but unfortunately it's not referenced in the `package.json`.

```json
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
```

**Comment by warrenseine** [comment]
Indeed, I was thinking about `onfido.min.js`. You make a good point, I'll try to understand why the package points to `index.js`.

Dependencies such as `@onfido/active-video-capture` or `@onfido/opencv` are bundled into `/dist/onfido.activeVideo.min.js` and `/dist/onfido.documentAuto.min.js`, which are expected to be lazy-loaded. If everything gets rebundled together, I understand why it's 10 MB.

Let me sync with the team and get back to you.

**Comment by giomogna** [comment]
I tried to patch the `package.json` (with [`patch-package`](https://github.com/ds300/patch-package) or [`pnpm`](https://pnpm.io/cli/patch))  and the bundle size indeed dropped to a total of ~2.4MB (styles excluded)

```diff
-  "main": "lib/index.js",
+  "main": "dist/onfido.min.js",
```

Other than that the only change I had to make was importing manually the styles
```ts
import "onfido-sdk-ui/dist/style.css";
```

**Comment by ben-propflo** [comment]
Hey @warrenseine is there any update on this?

**Comment by warrenseine** [comment]
Hello @ben-propflo, yep, we're on it. It has non-trivial ramifications so we cannot give an ETA.

**Comment by ArneSchoonvliet** [comment]
Any update?

**Comment by warrenseine** [comment]
Yes, since 12.3.0, you can use a [split version of the bundle](https://github.com/onfido/onfido-sdk-ui#split-bundle). Please test it and tell us if that helps.
